repos/system_upgrade/common/actors/addupgradebootentry/libraries/addupgradebootentry.py

```diff
@@ -51,11 +51,16 @@ def _remove_old_upgrade_boot_entry(kernel_dst_path, configs=None):
         '/usr/sbin/grubby',
         '--remove-kernel', '{0}'.format(kernel_dst_path)
     ]
-    if configs:
-        for config in configs:
-            run(cmd + ['-c', config])
-    else:
-        run(cmd)
+    try:
+        if configs:
+            for config in configs:
+                run(cmd + ['-c', config])
+        else:
+            run(cmd)
+    except CalledProcessError as e:
+        api.current_logger().warning(
```

Comment:
I would put here TODO  (or above) to detect whether the entry exists or not; so users will not be afraid by the message when it's irrelevant.

Reply:
I've added the todo and removed the unused variable.